### PR TITLE
VP-4570: Fix folder duplication when update it's name

### PR DESCRIPTION
**Pull request by AliveMen**
### Problem
The blog folder is duplicated when update folder name

### Solution
1. Create new folder 
2. Copy content
3. Remove old one 

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

File: src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js [hunk 1]
```diff
@@ -1,200 +1,227 @@
...
+                    if (!parentUrl) {
+                        contentApi.createFolder(
+                            { contentType: blade.contentType, storeId: blade.storeId },
+                            { name: folderName, parentUrl: parentUrl },
+                            $scope.saveWithMetadata
+                        );
+                    } else {
+                        if (blade.origEntity.name !== blade.currentEntity.name) {
+                            let urlsToDelete = [];
+                            let oldFilename = blade.origEntity.name + '.md';
```
Comment (pushnitsa) [inline src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js:90]: Is it ok have hardcoded file extension? We always having only "md" files?

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js (hunk 1 above)
Comment (AliveMen) [reply src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js]: moved to constant

File: src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js [hunk 2]
```diff
@@ -1,200 +1,227 @@
...
+                    }
+                );
+
+                function getBlogBlobName() {
+                    return blade.currentEntity.name + '/' + blade.currentEntity.name + '.md'
+                }
 
-    $scope.languages = settings.getValues({ id: 'VirtoCommerce.Core.General.Languages' });
+                function isDirty() {
+                    return !angular.equals(blade.currentEntity, blade.origEntity) && blade.hasUpdatePermission();
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js:198]: Why you want to move hasUpdatePermission in this case? In all other cases hasUpdatePermission always at the end of line.

File: src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js [hunk 3]
```diff
@@ -1,200 +1,227 @@
...
+
+                        //        dialogService.showConfirmationDialog(dialog);
+                        //    },
+                        //    canExecuteMethod: function () { return true; },
+                        //    permission: blade.updatePermission
+                        //}
+                    ];
+                }
+
+                blade.toolbarCommands = blade.toolbarCommands || [];
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js:174]: Yes, it possible. Where is a profit?

File: src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js [hunk 4]
```diff
@@ -1,200 +1,227 @@
...
+                        {
+                            name: "platform.commands.reset",
+                            icon: 'fa fa-undo',
+                            executeMethod: function() {
+                                angular.copy(blade.origEntity, blade.currentEntity);
+                            },
+                            canExecuteMethod: isDirty,
+                            permission: blade.updatePermission
+                        }
+                        //{
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js]: Removed

File: src/VirtoCommerce.ContentModule.Web/Scripts/resources/contentResource.js [hunk 5]
```diff
@@ -116,6 +116,7 @@
         unpack: { url: 'api/content/:contentType/:storeId/unpack' },
         createFolder: { url: 'api/content/:contentType/:storeId/folder', method: 'POST' },
         copy: { url: 'api/content/copy' },
-        move: { url: 'api/content/:contentType/:storeId/move' }
+        move: { url: 'api/content/:contentType/:storeId/move' },
+        delete: {url: 'api/content/:contentType/:storeId', method: 'DELETE'}
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply src/VirtoCommerce.ContentModule.Web/Scripts/resources/contentResource.js]: File reformated

**Review by pushnitsa — APPROVED** [review]

**Review by yecli — APPROVED** [review]

**Pull request merged**